designs/plugins-phase-2.md

```diff
@@ -0,0 +1,105 @@
+# Plugins Phase 2
+
+## Overview
+
+Phase 2 of the scaffold plugin system adds support for chaining plugins together. The idea for plugins phase 2 is to have support for a plugin architecture that enables plugins to be separate binaries discoverable by the Kubebuilder CLI binary via user specified plugin file paths and that they are language indepedent. Some of the phase 2 high-level requirements summarized from the feature request and discussions are: 
```

Comment:
Interpreted languages, i.e. Python, do not generate binaries. Are we leaving them apart intentionally? I think that interpreted languages are well suited for this kind of scripting purposes and considering them an option would be interesting.